src/client/telemetry/index.ts

```diff
@@ -314,7 +314,11 @@ export function captureTelemetry<This, P extends IEventNamePropertyMapping, E ex
                 return properties;
             };
 
+            // Determine if this is the first time we're sending this telemetry event for this same (class/method).
             const stopWatch = captureDuration ? new StopWatch() : undefined;
+            const key = `${eventName.toString()}${JSON.stringify(props() || {})}`;
+            const firstTime = !timesSeenThisEventWithSameProperties.has(key);
+            timesSeenThisEventWithSameProperties.add(key);
```

Comment:
Generic way to determine whether we've already sent this telemetry event before or not.